core/commonMain/src/ArgType.kt

```diff
@@ -81,6 +81,36 @@ abstract class ArgType<T : Any>(val hasParameter: kotlin.Boolean) {
             if (value in values) value
             else throw ParsingException("Option $name is expected to be one of $values. $value is provided.")
     }
+
+    companion object {
+        /**
+         * Helper for arguments that have limited set of possible values represented as enumeration constants.
+         */
+        inline fun <reified T: Enum<T>> EnumChoice(): EnumArgChoiceImpl<T> {
+            return EnumArgChoiceImpl(enumValues())
+        }
+    }
+
+    class EnumArgChoiceImpl<T: Enum<T>>(choices: Array<T>): ArgType<T>(true)
+    {
+        private val choicesMap: Map<kotlin.String, T> = choices.associateBy { it.toString() }
```

Comment:
Optional improvement suggestion: provide to/from conversion functions externally.